app/routes.py

```diff
@@ -1 +1,275 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+import requests
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details":"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"details": f"{cls.__name__} {model_id} is not found"}, 404))
+    
+    return model
+
+def validate_request(cls, request):
+    request_body = request.get_json()
+
+    if not request_body.get("title"):
+        abort(make_response({"details":"Invalid data"}, 400))
+    
+    elif not request_body.get("description") and cls == Task:
+        abort(make_response({"details":f"Invalid data"}, 400))
+    
+    return request_body
+  
+
+@task_bp.route("", methods=["POST"])
+def create_task():
+    request_body = validate_request(Task, request)
+
+    new_task = Task.from_dict(request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task":new_task.to_dict()}, 201
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+
+    sort_query = request.args.get("sort", "asc")
+    title_query = request.args.get("title")
+    description_query = request.args.get("description")
+   
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+
+    elif title_query:
+        tasks = Task.query.filter_by(title = title_query)
```

Comment:
Nice work on getting the database to do most of the lifting when it comes to sorting the tasks